src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookConsumer.java

```diff
@@ -98,7 +73,11 @@ private void triggerJob(RefsChangedWebhookEvent event,
 
     private static Collection<? extends SCM> getScms(ParameterizedJobMixIn.ParameterizedJob<?, ?> job) {
         SCMTriggerItem triggerItem = SCMTriggerItem.SCMTriggerItems.asSCMTriggerItem(job);
-        if (triggerItem != null) {
+        if (triggerItem instanceof WorkflowJob) {
+            if (((WorkflowJob) triggerItem).getDefinition() instanceof CpsScmFlowDefinition) {
+                return Collections.singletonList(((CpsScmFlowDefinition) ((WorkflowJob) triggerItem).getDefinition()).getScm());
```

Comment:
/nit - You should log if the above expectation of WorkflowJob being `CpsScmFlowDefinition` is not met.

Reply:
This is trixy, I guess you're right, but if I use a pipeline script in Jenkins I won't have a `CpsScmFlowDefinition` which makes this task impossible, but maybe that's all the more reason to log, so yes, I agree let's log it

Reply:
Yeah I think it's a warning "can't create a webhook on a pipeline with no SCM" or similar